src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -308,66 +308,71 @@ public static void makeGenotypeCall(final int ploidy,
                                         final List<Allele> allelesToUse,
                                         final List<Allele> originalGT,
                                         final GenotypePriorCalculator gpc) {
-        if(originalGT == null && assignmentMethod == GenotypeAssignmentMethod.BEST_MATCH_TO_ORIGINAL) {
-            throw new IllegalArgumentException("original GT cannot be null if assignmentMethod is BEST_MATCH_TO_ORIGINAL");
-        }
-        if (assignmentMethod == GenotypeAssignmentMethod.SET_TO_NO_CALL) {
-            gb.alleles(noCallAlleles(ploidy));
-        } else if (assignmentMethod == GenotypeAssignmentMethod.USE_PLS_TO_ASSIGN ||
-                    assignmentMethod == GenotypeAssignmentMethod.PREFER_PLS) {
-            if ( genotypeLikelihoods == null || !isInformative(genotypeLikelihoods) ) {
-                if (assignmentMethod == GenotypeAssignmentMethod.PREFER_PLS) {
-                    if (originalGT == null) {
-                        throw new IllegalArgumentException("original GT cannot be null if assignmentMethod is PREFER_PLS");
-                    } else {
-                        gb.alleles(bestMatchToOriginalGT(allelesToUse, originalGT));
-                    }
+        switch (assignmentMethod) {
+            case SET_TO_NO_CALL -> gb.alleles(noCallAlleles(ploidy));
+            case SET_TO_NO_CALL_NO_ANNOTATIONS -> gb.alleles(noCallAlleles(ploidy)).noGQ().noAD().noPL().noAttributes();
+            case PREFER_PLS -> {
+                if (!isInformative(genotypeLikelihoods)) {
+                    Utils.nonNull(originalGT, "original GT cannot be null if assignmentMethod is " + GenotypeAssignmentMethod.PREFER_PLS);
+                    gb.alleles(bestMatchToOriginalGT(allelesToUse, originalGT));
                 } else {
-                    gb.alleles(noCallAlleles(ploidy)).noGQ();
+                    setUsingPls(ploidy, gb, genotypeLikelihoods, allelesToUse);
                 }
-            } else {
-                final int maxLikelihoodIndex = MathUtils.maxElementIndex(genotypeLikelihoods);
-                final GenotypeAlleleCounts alleleCounts = GenotypesCache.get(ploidy, maxLikelihoodIndex);
-                final List<Allele> finalAlleles = alleleCounts.asAlleleList(allelesToUse);
-                if (finalAlleles.contains(Allele.NON_REF_ALLELE)) {
-                    final Allele ref = allelesToUse.stream().filter(Allele::isReference).collect(Collectors.toList()).get(0);
-                    gb.alleles(Collections.nCopies(ploidy, ref));
-                    gb.PL(new int[genotypeLikelihoods.length]).log10PError(0);
+            }
+            case USE_PLS_TO_ASSIGN -> {
```

Comment:
I separated these PREFER_PLS and USE_PLS_TO_ASSIGN so it's clear where they're different and pulled out the shared setUsingPls.  That could probably use a better name.